torch/_inductor/scheduler.py

```diff
@@ -190,6 +188,14 @@ def prune_deps(self):
             if dep.name not in self.scheduler.available_buffer_names
         }
 
+    def prune_weak_deps(self):
```

Comment:
is there another spot where deps are removed after dce? I'm curious if we need to handle weakdeps specially here

Reply:
Not that I'm aware of. It wouldn't really make sense because usually if there's a dependency then the node isn't DCE'd.